### PR TITLE
Delete deprecated three integrations, `skopt`, `catalyst`, and `fastaiv1`

### PR DESCRIPTION
**Pull request by eukaryo**
## Motivation

I want to delete three deprecated integration modules.

This PR is related to https://github.com/optuna/optuna/pull/5407 and https://github.com/optuna/optuna-examples/pull/256 .

## Description of the changes

I modified or deleted three py files, `__init__.py`, `README.md`, `pyproject.toml`, and `index.rst`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,21 +19,18 @@ Optuna-Integration API reference is [here](https://optuna-integration.readthedoc
...
 * [CatBoost](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#catboost)  ([example](https://github.com/optuna/optuna-examples/blob/main/catboost/catboost_pruning.py))
 * [Chainer](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#chainer)  ([example](https://github.com/optuna/optuna-examples/tree/main/chainer/chainer_integration.py))
 * [ChainerMN](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#chainermn)  ([example](https://github.com/optuna/optuna-examples/tree/main/chainer/chainermn_simple.py))
 * [Dask](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#dask) ([example](https://github.com/optuna/optuna-examples/tree/main/dask/dask_simple.py))
 * [FastAI](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#fast-ai) (
-  V1 ([example](https://github.com/optuna/optuna-examples/tree/main/fastai/fastaiv1_simple.py)),
   V2  ([example](https://github.com/optuna/optuna-examples/tree/main/fastai/fastaiv2_simple.py)))
```
Comment (nabenabe0928) [inline README.md:27]:
```md
* [FastAI](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#fast-ai) ([example](https://github.com/optuna/optuna-examples/tree/main/fastai/fastaiv2_simple.py))
```

File: pyproject.toml [hunk 2]
```diff
@@ -58,7 +58,6 @@ document = [
 ]
 all = [
   "botorch",
-  "catalyst",
```
Comment (nabenabe0928) [inline pyproject.toml:61]: Could you add a version constraint to fastai?

**Review by nabenabe0928 — CHANGES_REQUESTED** [review]
Could you also remove CmaEsSampler?

File: pyproject.toml (hunk 2 above)
Comment (eukaryo) [reply pyproject.toml:61]: Done in commit https://github.com/optuna/optuna-integration/pull/114/commits/07ed516dd27e97c4bf8d9c7be15ade08ca124f72 .

File: README.md (hunk 1 above)
Comment (eukaryo) [reply README.md]: Done in commit https://github.com/optuna/optuna-integration/pull/114/commits/d8c59769f9884958debd371d37cf0efc6c8579d2 .

**Review by nabenabe0928 — APPROVED** [review]
LGTM!
Followup:
- Remove CmaesSampler

**Pull request merged**